### PR TITLE
Bug: specify axis=1 when reducing sum

### PR DESCRIPTION
**Pull request by andySigler**
`tf.reduce_sum()` with `axis=None` returns a scalar, so that the following `tf.reduce_mean()` does nothing but return that same scalar. I seems the intention here was for the sum to be reduced on a per-example basis, before then finding the mean across all examples.

### CODE REVIEW

**Comment by andySigler** [comment]
nvm, understand gradient tape better now
